### PR TITLE
make sure buffer doesn't contain garbage

### LINKED ISSUE #132: Sound repeats forever when detaching sink (on Windows)
On Linux, if I detach the sink, it keeps playing the sound clip until it finishes. On Windows, a short segment (<100 ms) of the sound clip I'm playing starts repeating forever when I detach the sink. I've even tried stopping the sink before detaching it, but it has no effect. The only workaround I've found is to sleep for some time (e.g. 100 ms) after stopping it, before detaching it, but that's a bit inelegant.

```
extern crate rodio;

use std::fs::File;
use std::io::BufReader;
use std::time::Duration;

fn main() {
	let endpoint = rodio::get_default_endpoint().unwrap();
	let mut sink = rodio::Sink::new(&endpoint);
	sink.set_volume(0.1);

        // Using beep.wav from rodio/examples
	let source = File::open("res/beep.wav").ok()
		.and_then(|file| rodio::Decoder::new(BufReader::new(file)).ok()).unwrap();

	sink.append(source);
	std::thread::sleep(Duration::from_millis(100));

	// A stop, followed by a short sleep, prevents the bug from occuring.
	// sink.stop();
	// std::thread::sleep(Duration::from_millis(100));

	sink.detach();
	std::thread::sleep(Duration::from_secs(10)); // Sleep so we can hear the annoying sound for a while.
}
```

### PR DESCRIPTION
**Pull request by chyvonomys**
This fixes #132.

Given `buffer` is obtained from `IAudioRenderClient::GetBuffer` and may contain garbage (previous sound for example). `IAudioRenderClient::ReleaseBuffer` currently always returns whole buffer (in `cpal`) no matter if we filled it completely or partially or none, so we have to be sure it is filled with correct data.

My test code:
```rust
extern crate rodio;

use std::time::Duration;
use rodio::Source;

fn main() {
    let endpoint = rodio::get_default_endpoint().unwrap();
    let mut sink = rodio::Sink::new(&endpoint);
    sink.set_volume(0.1);

    let source = rodio::source::SineWave::new(440).take_duration(Duration::from_secs(10));

    println!("start 10 sec of sound");
    sink.append(source);

    std::thread::sleep(Duration::from_secs(1));

    println!("detach after 1 sec");
    sink.detach();

    std::thread::sleep(Duration::from_secs(9));

    println!("silence");

    std::thread::sleep(Duration::from_secs(100));
}
```

### CODE REVIEW

**Comment by tomaka** [comment]
Thanks!
